mmcv/cnn/utils/weight_init.py

```diff
@@ -146,7 +150,8 @@ def init(m):
                 constant_init(m, self.val, self.bias)
             else:
                 layername = m.__class__.__name__
-                if layername in self.layer:
+                basesname = _get_bases_name(m)
+                if bool(set(self.layer) & set([layername] + basesname)):
```

Comment:
```suggestion
                if len(set(self.layer) & set([layername] + basesname)):
```
maybe be more intuitive.